src/utils/is.ts

```diff
@@ -7,17 +7,28 @@ const isElementHidden = (computedStyle: CSSStyleDeclaration): boolean => {
   );
 };
 
-export const isVisible = (node: HTMLElement | undefined): boolean =>
-  !node ||
-  // @ts-ignore
-  node === document ||
-  (node && node.nodeType === Node.DOCUMENT_NODE) ||
-  (!isElementHidden(window.getComputedStyle(node, null)) &&
-    isVisible(
-      node.parentNode && node.parentNode.nodeType === Node.DOCUMENT_FRAGMENT_NODE
-        ? (node.parentNode as any).host
-        : node.parentNode
-    ));
+export const isVisible = (node: HTMLElement | undefined, visibilityCache: Map<HTMLElement, boolean>): boolean => {
+    if(!node) {
+        return false;
+    }
+    const cached = visibilityCache.get(node)
+    if(cached) {
```

Comment:
that is working only for "true" cases, while false is legit as well
```suggestion
    if (cached!==undefined) {
```